src/vivarium/framework/results/context.py

```diff
@@ -105,9 +105,10 @@ def gather_results(self, population: pd.DataFrame, event_name: str) -> Dict[str,
         ].items():
             # Results production can be simplified to
             # filter -> groupby -> aggregate in all situations we've seen.
+            stratified_pop = population.copy()
             if pop_filter:
```

Comment:
Wouldn't something like this be a bit cleaner:
```
if pop_filter:
    filtered_pop = population.query(pop_filter)
else:
    filtered_pop = population
```

That way we don't have to copy the pop table and in the no filter case we don't use any extra memory.

Reply:
That's a good point. That copy could be expensive and is ultimately unnecessary.